### PR TITLE
Fixed the issue of appcrash in DayFragmentActivty

### PR DESCRIPTION
**Pull request by sridharjajoo**
Fixed the issue related to app crash #134

### CODE REVIEW

**Comment by yatna** [comment]
@sridharjajoo As asked in the issue, you had to specify the reason for the crash before sending the PR. Anyways, you can do it now.

**Comment by sridharjajoo** [comment]
@yatna Sorry for my mistake .

The following error was shown in the logcat:

FATAL EXCEPTION: main
Process: com.peacecorps.malaria, PID: 15477
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=196610, result=0, data=null} to activity {com.peacecorps.malaria/com.peacecorps.malaria.activities.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.compareTo(java.lang.String)' on a null object reference

The main problem was because of the lines:
if (cursor.getString(0).compareTo("yes") == 0) {
                    prevDate = cursor.getInt(2);
                    prevDateMonth = cursor.getInt(3);
                    if (Math.abs(currDate - prevDate) <= 1)
                        dosesInaRow++;
                }
Here the condition expects either yes or no and is not prepared for a case when neither of the options is selected i.e in case of a null object reference . That was the main reason for the crash.

**Comment by yatna** [comment]
@chhavip :+1: 
@sridharjajoo Good Work! After Chhavi approves and merges this PR, please close the related issue.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/db/DatabaseSQLiteHelper.java [hunk 1]
```diff
@@ -393,12 +393,14 @@ public int getDosesInaRowDaily()
                 } catch (Exception e) {
                     return 0;
                 }
+
+                if(cursor.getString(0)!=null){
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/db/DatabaseSQLiteHelper.java]: Add proper indentation to the inner if condition.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/db/DatabaseSQLiteHelper.java [hunk 2]
```diff
@@ -393,12 +393,14 @@ public int getDosesInaRowDaily()
...
                 }
+
+                if(cursor.getString(0)!=null){
                 if (cursor.getString(0).compareTo("yes") == 0) {
                     prevDate = cursor.getInt(2);
                     prevDateMonth = cursor.getInt(3);
                     if (Math.abs(currDate - prevDate) <= 1)
                         dosesInaRow++;
-                }
+                }}
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/db/DatabaseSQLiteHelper.java]: Next closing bracket on next line.

**Comment by sridharjajoo** [comment]
@chhavip Made the required changes.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/db/DatabaseSQLiteHelper.java [hunk 3]
```diff
@@ -393,11 +393,14 @@ public int getDosesInaRowDaily()
...
-                        dosesInaRow++;
+
+                if(cursor.getString(0)!=null){
+                    if (cursor.getString(0).compareTo("yes") == 0){
+                        prevDate = cursor.getInt(2);
+                        prevDateMonth = cursor.getInt(3);
+                        if (Math.abs(currDate - prevDate) <= 1)
+                            dosesInaRow++;
+                }
                 }
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/db/DatabaseSQLiteHelper.java]: The brackets should have a different level of closing as they are of different order.

**Comment by sridharjajoo** [comment]
@chhavip Kindly review it.

**Comment by chhavip** [comment]
Approved. Merging. Thank you for the work!

**Pull request merged**